WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/BuiltInCardReaderPaymentAlertsProvider.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+import Yosemite
+import MessageUI
+import enum Hardware.CardReaderServiceError
+import enum Hardware.UnderlyingError
+
+final class BuiltInCardReaderPaymentAlertsProvider: CardReaderTransactionAlertsProviding {
+    var name: String = ""
+    var amount: String = ""
+
+    func preparingReader(onCancel: @escaping () -> Void) -> CardPresentPaymentsModalViewModel {
+        CardPresentModalPreparingReader(cancelAction: onCancel)
+    }
+
+    func tapOrInsertCard(title: String,
+                         amount: String,
+                         inputMethods: Yosemite.CardReaderInput,
+                         onCancel: @escaping () -> Void) -> CardPresentPaymentsModalViewModel {
+        name = title
+        self.amount = amount
+        return CardPresentModalBuiltInFollowReaderInstructions(name: name,
+                                              amount: amount,
+                                              transactionType: .collectPayment,
+                                              inputMethods: inputMethods,
+                                              onCancel: { })
+    }
+
+    func displayReaderMessage(message: String) -> CardPresentPaymentsModalViewModel {
```

Comment:
I didn't do this because the built-in reader alerts will all change over the next few PRs... but when that's done I'll look to share code where I can.